test/code/plugins/oms_common_test.rb

```diff
@@ -290,6 +290,12 @@ def test_parse_json_record_encoding
       assert_equal("{\"DataItems\":[{\"Message\":\"iPhone®\"}]}", parsed_record, "parse json record utf-8 encoding failed");
     end	
     
+    def test_safe_dump_simple_hash_array
```

Comment:
Missing unit tests:
- When hash value is not a string
- When JSON.dump succeeds and no re-encoding is performed
- When reencoding fails and an error is logged